### PR TITLE
Fixed Unable to open failed Action

### PR DESCRIPTION
**Pull request by arvindgh**
Added exception handling for ProcessInputValueForDriver

### CODE REVIEW

File: Ginger/GingerCoreNET/Run/GingerRunner.cs [hunk 1]
```diff
@@ -1423,44 +1423,50 @@ public void ProcessReturnValueForDriver(Act act)
...
+                                IV.ValueForDriver = string.Empty;
+                            }
                         }
                     }
                 }
+                act.ValueExpression.DecryptFlag = true;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Exception occurred in ProcessInputValueForDriver : ", ex);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/GingerRunner.cs:1468]: 1- so only try-catch was added? if yes, please explain why it's enough and not need to handle the exception reason

2- maybe need to add the try-catch for each calculation in the list so exception happened only the specific value will

Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/GingerRunner.cs:1468]: The error message don't add anything, we know it's exception by adding ex object and we know it happen in ProcessInputValueForDriver from exception stackTrace which will be added to the log from ex object. Error message should be something like
Reporter.ToLog(eLogLevel.ERROR, string."Error occured while calculating the Action input VE's", ex);

File: Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs [hunk 2]
```diff
@@ -96,7 +96,7 @@ public class ValueExpression : IValueExpression
         public static Regex rxEnvUrlPattern = new Regex(@"{(\bEnvURL App=)\w+\b[^{}]*}", RegexOptions.Compiled);
         
         private static Regex VBSRegex = new Regex(@"{[V|E|VBS]" + rxVar + "[^{}]*}", RegexOptions.Compiled);
-        private static Regex rxe = new Regex(@"{RegEx" + rxVare + ".*}", RegexOptions.Compiled);
+        private static Regex rxe = new Regex(@"{RegEx" + rxVare + ".*}", RegexOptions.Singleline);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs:99]: why "Singleline" option is insted "Compiled" option? can't we have both?
like "RegexOptions.Compiled | RegexOptions.Singleline"?

Comment (arvindgh) [reply Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs:99]: Yes, Updated.

File: Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs [hunk 3]
```diff
@@ -96,7 +96,7 @@ public class ValueExpression : IValueExpression
         public static Regex rxEnvUrlPattern = new Regex(@"{(\bEnvURL App=)\w+\b[^{}]*}", RegexOptions.Compiled);
         
         private static Regex VBSRegex = new Regex(@"{[V|E|VBS]" + rxVar + "[^{}]*}", RegexOptions.Compiled);
-        private static Regex rxe = new Regex(@"{RegEx" + rxVare + ".*}", RegexOptions.Compiled);
+        private static Regex rxe = new Regex(@"{RegEx" + rxVare + ".*}", RegexOptions.Compiled | RegexOptions.Singleline);
         private static Regex rfunc = new Regex("{Function(\\s)*Fun(\\s)*=(\\s)*([a-zA-Z]|\\d)*\\((\")*([^\\)}\\({])*(\")*\\)}", RegexOptions.Compiled);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs:100]: please add unit tests, with and without new line to make sure Regex works as we expected

**Comment by MeniKadosh1** [comment]
Arvind please check why Windows unit tests are failing

**Pull request merged**